Makefile

```diff
@@ -204,3 +204,46 @@ migrate: migrate-lms migrate-cms
 # Part of https://github.com/openedx/wg-developer-experience/issues/136
 ubuntu-requirements: ## Install ubuntu 22.04 system packages needed for `pip install` to work on ubuntu.
 	sudo apt install libmysqlclient-dev libxmlsec1-dev
+
+eslint:	## check javascript for quality issues
+	python scripts/eslint.py
+
+xsslint: ## check xss for quality issuest
+	python scripts/xsslint/xss_linter.py \
+	--rule-totals \
+	--config=scripts.xsslint_config \
+	--thresholds=scripts/xsslint_thresholds.json
+
+pycodestyle: ## check python files for quality issues 
+	pycodestyle .
+
+## Re-enable --lint flag when this issue https://github.com/openedx/edx-platform/issues/35775 is resolved
+pii_check: ## check django models for pii annotations
+	DJANGO_SETTINGS_MODULE=cms.envs.test \
+	code_annotations django_find_annotations \
+		--config_file .pii_annotations.yml \
+		--app_name cms \
+		--coverage \
+		--lint
+	
+	DJANGO_SETTINGS_MODULE=lms.envs.test \
+	code_annotations django_find_annotations \
+		--config_file .pii_annotations.yml \
+		--app_name lms \
+		--coverage \
+		--lint	
+
+check_keywords: ## check django models for reserve keywords
+	DJANGO_SETTINGS_MODULE=cms.envs.test \
+	python manage.py cms check_reserved_keywords \
+	--override_file db_keyword_overrides.yml
+
+	DJANGO_SETTINGS_MODULE=lms.envs.test \
+	python manage.py lms check_reserved_keywords \
+	--override_file db_keyword_overrides.yml
+
+test-js: ## run javascript tests
+	node --max_old_space_size=4096 node_modules/.bin/karma start common/static/karma_common.conf.js \
+		--single-run=true \
+		--capture-timeout=60000 \
+		--browsers=FirefoxNoUpdates
```

Comment:
This only runs a single JS test suite (CMS). Here's what I've come up with to run all of them:

```suggestion
NODE_KARMA_START=node --max_old_space_size=4096 node_modules/.bin/karma start --single-run=true --capture-timeout=60000 --browsers=FirefoxNoUpdates

test-js: ## run javascript tests
	TEST_SUITE='cms javascript' $(NODE_KARMA_START) cms/static/karma_cms.conf.js
	TEST_SUITE='cms-squire javascript' $(NODE_KARMA_START) cms/static/karma_cms_squire.conf.js
	TEST_SUITE='cms-webpack javascript' $(NODE_KARMA_START) cms/static/karma_cms_webpack.conf.js
	TEST_SUITE='lms javascript' $(NODE_KARMA_START) lms/static/karma_lms.conf.js
	TEST_SUITE='xmodule javascript' $(NODE_KARMA_START) xmodule/js/karma_xmodule.conf.js
	TEST_SUITE='xmodule-webpack javascript' $(NODE_KARMA_START) xmodule/js/karma_xmodule_webpack.conf.js
	TEST_SUITE='common javascript' $(NODE_KARMA_START) common/static/karma_cms.conf.js
	TEST_SUITE='common-requirejs javascript' $(NODE_KARMA_START) common/static/karma_cms_requirejs.conf.js
```

I've added this to my [validation PR](https://github.com/kdmccormick/edx-platform/pull/20). I'm still seeing failures that I haven't figured out yet, but at least I see that all suites are failing rather than just one of them.

Reply:
@kdmccormick should i integrate #35957 instead of using these Make commands?

Reply:
@salman2013 , when you integrate that, keep them webpack-based tests in package.json, but remove them from `test-cms`, `test-lms`, and `test-xmodule` (which will remove from `npm run test`). That way, developers can try running webpack tests locally, but they will not block CI. Unfortunately, there is no way to leave an explanatory comment in package.json, but we can include an explanation when we eventually update [testing.rst](https://github.com/openedx/edx-platform/issues/35956).

Reply:
@kdmccormick I did not get the exact idea about `but remove them from test-cms, test-lms, and test-xmodule (which will remove from npm run test).` could you please elaborate more on this?

Reply:
Sure. In case you are not familiar, the `"scripts"` dictionary in `package.json` acts similar to a Makefile. Each key is an `npm run` command target, and the value is the the underlying command. For example, `npm run watch-sass` causes the `"watch-sass"` script to be executed, which is defined as `scripts/watch_sass.sh`. Some of the definitions invoke other `npm run` commands, for example:

    "build": "npm run webpack && npm run compile-sass",

in https://github.com/openedx/edx-platform/pull/35957, I have made it so that `npm run test` invokes sub-commands for each of LMS, CMS, Common, and XModule:

    "test": "npm run test-cms && npm run test-lms && npm run test-xmodule && npm run test-common && npm run test-jest",

And each of those invokes further sub-commands for its suites. For example, CMS has three suites:

    "test-cms": "npm run test-cms-vanilla && npm run test-cms-require && npm run test-cms-webpack",

Since [the `-webpack` suites are broken](https://github.com/openedx/edx-platform/commit/4839b754674a2d2c44400b340ac2ec4a1383e547), though, you will need to omit them from these sub-commands.

I should note that `test-lms` *only* invokes a `-webpack` suite, so removing the webpack suite from it will leave it empty. Perhaps you could replace it with something like:

    "test-lms": "echo 'WARNING: Webpack JS tests are disabled. No LMS JS tests will be run. See https://github.com/openedx/edx-platform/issues/35956 for details.''",

Reply:
got it, thanks for the detail.

Reply:
no problem